### PR TITLE
Add happy path test for searchSubmission method

### LINKED ISSUE #301: Write happy path test for searchSubmissions
# Summary
We currently don't have any tests to verify that searchSubmissions is working as intended.

# Requirements

- [x] Write a happy path test that verifies searchSubmissions is working correctly.
- [x] Ensure that the doc strings for searchSubmissions is filled.

### PR DESCRIPTION
**Pull request by Anissa3005**
Resolves #301 

# What changed

1. Add happy path test for searchSubmission method.
2. Add tests for the different filters.
3. Add docstrings to the searchSubmission method.
4. Add test environment for the firestore emulator. After every test the emulator will now clear the data.

# To do

1. Add error path for searchSubmission
2. Add test enviroments to server.tests and facilities.test.

### CODE REVIEW

File: __tests__/submissions.test.ts [hunk 1]
```diff
@@ -259,3 +261,297 @@ describe('getSubmissionById', () => {
...
+    let testEnv: RulesTestEnvironment
+
+    const server = new ApolloServer({
+        typeDefs: loadSchema(),
+        resolvers
+    })
+
+    beforeAll(async () => {
+        ({ url } = await startStandaloneServer(server, {listen: { port: 0 }}))
+        // emulator = await initiatilizeFirebaseInstance()
```
Comment (theyokohamalife) [inline __tests__/submissions.test.ts:277]: Don't forget to delete this

File: __tests__/submissions.test.ts [hunk 2]
```diff
@@ -164,19 +178,31 @@ describe('getSubmissionById', () => {
 
     beforeAll(async () => {
         ({ url } = await startStandaloneServer(server, {listen: { port: 0 }}))
-        await initiatilizeFirebaseInstance()
+        testEnv = await initializeTestEnvironment({
+            projectId: process.env.FIRESTORE_PROJECT_ID,
+            firestore: {
+                rules: fs.readFileSync('./firestore.rules', 'utf8')
+            }
+        })
```
Comment (theyokohamalife) [inline __tests__/submissions.test.ts:186]: I'm wondering how we can refactor `testEnv` so it doesn't need to be defined for every test. 🤔

File: __tests__/submissions.test.ts [hunk 3]
```diff
@@ -203,20 +229,19 @@ describe('getSubmissionById', () => {
...
-        expect(submissionResponse.isRejected).toBe(createdSubmission.isRejected)
+        expect(submissionResponse.googleMapsUrl).toBe(createdSubmissionData.googleMapsUrl)
+        expect(submissionResponse.healthcareProfessionalName).toBe(createdSubmissionData.healthcareProfessionalName)
+        expect(submissionResponse.spokenLanguages).toEqual(createdSubmissionData.spokenLanguages)
+        expect(submissionResponse.isApproved).toBe(createdSubmissionData.isApproved)
+        expect(submissionResponse.isUnderReview).toBe(createdSubmissionData.isUnderReview)
+        expect(submissionResponse.isRejected).toBe(createdSubmissionData.isRejected)
     })
 
     it('get an error when submission does not exist', async () => {
```
Comment (theyokohamalife) [inline __tests__/submissions.test.ts:247]:
```suggestion
    it('returns an error when submission does not exist', async () => {
```

File: __tests__/submissions.test.ts [hunk 4]
```diff
@@ -248,14 +273,309 @@ describe('getSubmissionById', () => {
             }
         }
 
-        const submission = await request(url).post('/').send(submissionQuery)
+        // Get error data
+        const searchData = await request(url).post('/').send(submissionQuery)
 
         // Compare the data returned in the response to the updated fields that were sent
-        const submissionErrorResponse = submission.body
+        const submissionErrorResponse = searchData .body
```
Comment (theyokohamalife) [inline __tests__/submissions.test.ts:280]: You seem to have an empty space between each of the `searchData .` 
```suggestion
        const submissionErrorResponse = searchData.body
```

File: __tests__/submissions.test.ts [hunk 5]
```diff
@@ -248,14 +273,309 @@ describe('getSubmissionById', () => {
...
+        expect(resultData.healthcareProfessionalName).toBe(createdSubmissionData.healthcareProfessionalName)
+        expect(resultData.isApproved).toBe(createdSubmissionData.isApproved)
+        expect(resultData.isRejected).toBe(createdSubmissionData.isRejected)
+        expect(resultData.isUnderReview).toBe(createdSubmissionData.isUnderReview)
+        expect(resultData.spokenLanguages).toStrictEqual(createdSubmissionData.spokenLanguages)
+    })
+
+    it('get submissions using multiple filters combining isApproved, isRejected and underReview', 
+       async () => {
+           // In this test beside testing the filters I want to test if it excepts multiple filters
```
Comment (theyokohamalife) [inline __tests__/submissions.test.ts:482]:
```suggestion
           // In this test beside testing the filters I want to test if it excepts multiple filters
```
```suggestion
```

File: src/services/submissionService.ts [hunk 6]
```diff
@@ -56,6 +56,13 @@ function validateIdInput(id: string): Result<gqlTypes.Submission> {
     return validationResults
 }
 
+/**
+ * Get all the submissions in the database when no filters are provided.
+ * When there are filters provided it will return all the submissions according to the filters.
+ * @param filters An object that contains parameters to filter on. 
+ * When no parameters are provided, filters is an empty object.
+ * @returns A submissions object
+ */
```
Comment (theyokohamalife) [inline src/services/submissionService.ts:65]: Nice! 😎

**Review by ermish — APPROVED** [review]
@theyokohamalife and I added the code review changes so we can merge your changes in. 
Nice work! 🙌

**Pull request merged**